it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,312 @@
+
+  @Captor private ArgumentCaptor<Iterable<Mutation>> writeMutationCaptor;
+  @Captor private ArgumentCaptor<Iterable<String>> statementCaptor;
+  @Captor private ArgumentCaptor<String> instanceIdCaptor;
+  @Captor private ArgumentCaptor<String> databaseIdCaptor;
+
+  @Before
+  public void setUp() {
+    testManager = new DefaultSpannerResourceManager(spanner, testId, projectId, region);
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateInstanceFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getInstanceAdminClient().createInstance(any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().createDatabase(any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerUpdateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().updateDatabaseDdl(any(), any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_worksWhenSpannerDoesntThrowAnyError() {
+    //   arrange
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act
+    testManager.createTable(statement);
+
+    // assert
+    verify(spanner.getDatabaseAdminClient())
+        .updateDatabaseDdl(
+            argThat(s -> s.startsWith("test-id") && !s.equals("test-id")),
+            eq("test_id"),
```

Comment:
I only know what "test_id" is because I know the implementation. Someone less familiar wouldn't know where this value came from. It's probably best for these base tests to have the test id not have any special characters and pass in the constant for matching. Later we can have tests specifically to make sure characters were replaced and introduce variables for just those tests.